scipy/stats/setup.py

```diff
@@ -28,8 +36,24 @@ def configuration(parent_package='',top_path=None):
         sources=['mvn.pyf','mvndst.f'],
     )
 
-    return config
+    # add BiasedUrn module
+    config.add_data_files('biasedurn.pxd')
+    ext = config.add_extension(
+        'biasedurn',
+        sources=[
+            'biasedurn.cxx',
+            'biasedurn/impls.cpp',
+            'biasedurn/fnchyppr.cpp',
+            'biasedurn/wnchyppr.cpp',
+            'biasedurn/stoc1.cpp',
+            'biasedurn/stoc3.cpp'],
+        include_dirs=[np.get_include()],
+        define_macros=[('R_BUILD', None)],
+        language='c++',
+    )
```

Comment:
With the understanding that we want both C++14 and minimal modifications to the C++ sources, the easiest thing to do here is ignore the errors macOS is currently complaining about:
```python
extra_compile_args=['-Wno-narrowing'] if platform.system() == 'Darwin' else [],
```

Reply:
so just add that as an argument to `config.add_extension`?

Reply:
Yes, you'll also have to import the `platform` library unless there's a better scipy-centric way of figuring out what platform it's on